File: spec-traits/src/main/java/bsp/traits/JsonNotificationTrait.java
package bsp.traits;

import software.amazon.smithy.model.SourceLocation;
import software.amazon.smithy.model.shapes.ShapeId;
import software.amazon.smithy.model.traits.StringTrait;

public final class JsonNotificationTrait extends StringTrait {

	public static final ShapeId ID = ShapeId.from("jsonrpc#jsonNotification");

	public JsonNotificationTrait(String value, SourceLocation sourceLocation) {
		super(ID, value, sourceLocation);
	}

	public JsonNotificationTrait(String value) {
		this(value, SourceLocation.NONE);
	}

	public static final class Provider extends StringTrait.Provider<JsonNotificationTrait> {
		public Provider() {
			super(ID, JsonNotificationTrait::new);
		}
	}
}

File: spec-traits/src/main/kotlin/traits/JsonNotificationTrait.kt
package traits

import software.amazon.smithy.model.SourceLocation
import software.amazon.smithy.model.shapes.ShapeId
import software.amazon.smithy.model.traits.StringTrait

class JsonNotificationTrait(value: String, sourceLocation: SourceLocation) : StringTrait(ID, value, sourceLocation) {
    class Provider : StringTrait.Provider<JsonNotificationTrait>(ID, ::JsonNotificationTrait)
    companion object {
        val ID: ShapeId = ShapeId.from("traits#jsonNotification")
    }
}
